eo-maven-plugin/src/test/java/org/eolang/maven/rust_project/NamesTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.rust_project;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Test case for {@link Names}.
+ *
+ * @since 0.1
+ */
+final class NamesTest {
+    @Test
+    void solvesSameHashes(@TempDir final Path temp) throws IOException {
+        final Names dispatcher = new Names(temp);
+        final String one = "AaAaAa";
+        final String two = "AaAaBB";
+        MatcherAssert.assertThat(
+            one.hashCode(),
+            Matchers.equalTo(two.hashCode())
+        );
+        MatcherAssert.assertThat(
+            dispatcher.name(one),
+            Matchers.not(
+                dispatcher.name(two)
+            )
+        );
+    }
+
+    @Test
+    void recoversNames(@TempDir final Path temp) throws IOException {
+        final List<String> locations = IntStream.range(0, 1000)
+            .mapToObj(String::valueOf)
+            .collect(Collectors.toList());
+        final Names before = new Names(temp);
+        final Map<String, String> functions = locations.stream().collect(
+            Collectors.toMap(loc -> loc, loc -> before.name(loc))
+        );
+        MatcherAssert.assertThat(
+            locations.size(),
+            Matchers.equalTo(functions.size())
+        );
+        before.save();
+        final Names after = new Names(temp);
+        for (int iter = locations.size() - 1; iter >= 0; --iter) {
```

Comment:
@levBagryansky Do we need the reverse order here? It seems like we might not care about the order at all, or maybe I am missing something?

Reply:
@volodya-lombrozo  We realy need it in order to check that it names according to existing map and does not create new.

Reply:
@levBagryansky Please, change the code to the next:
```java
        for (final String loc : locations) {
            MatcherAssert.assertThat(
                functions.get(loc),
                Matchers.equalTo(after.name(loc))
            );
        }
```
And you will see that nothing is changed and test will pass.

Reply:
@volodya-lombrozo The test will pass but it will not check recovering anynore

Reply:
@volodya-lombrozo This test firstle creates the name base (via `before`) and then check it (`after` object). The goal is to check that `after` is built according to `before` so it must give name according `before` table. Thus, checking in sequancy order will not prove that names are already exist in table and are not created again

Reply:
@volodya-lombrozo the essence of the check is in the wrong order

Reply:
@levBagryansky I believe it's better to find an explicit way to check your assumption:

> prove that names are already exist in table

As you can see, the current solution rises lots of questions and not straightforward as could be. 
For example, you can get random item from `location` and just check it. No need to check all of them.

Reply:
@levBagryansky Maybe, but simple code is much better. Why do you think it's better to check every item?